### PR TITLE
improvements for Heavy Nukie Sword

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
The reflect component (like for csabers) was added to the power sword, but it never actually worked because the component required the item to actually be a csaber.
I changed the component to instead work with a proc on the item that returns whether or not it can currently reflect projectiles. If no proc is given (such as on the hadar itself) then the reflection is always on. (While blocking, of course.)

I also made the special attacks on the heavy sword (swipe and stab) have the colors of the corresponding stance on the sword.

https://user-images.githubusercontent.com/35579460/160827517-dbf856d7-338b-4d2b-abd6-9e142ce85562.mp4

## Why's this needed? 
It is good if features actually work. If this proves too strong, it can always be removed later.
This also allows adding of the reflect component to things that are not csabers, if desired.

It looks nicer if the special attacks have the correct color, as they already do on some other weapons like this.

## Changelog

```changelog
(u) zjdtmkhzt
(+) The nukeops heavy energy sword can now reflect projectiles like a csaber can. (This was always there, it just didn't work because it was bugged until now.)
```

### CODE REVIEW

File: code/datums/components/itemblock/saberblock.dm [hunk 1]
```diff
@@ -1,23 +1,27 @@
 //adds object properties to the sword while blocking with it.
 datum/component/itemblock/saberblock
 	bonus = 1 //bonus is a flag that determines whether or not the item tooltip will include "⛨ Block+: RESIST with this item for more info" when not blocking
+	var/can_block
```
Comment (pali6) [inline code/datums/components/itemblock/saberblock.dm:4]: can_block reads as a name of a boolean variable to me. Maybe `can_block_check` or `can_block_getter` or something like that?

Comment (zjdtmkhzt) [reply code/datums/components/itemblock/saberblock.dm:4]: Yup, that's a good point!

**Comment by Tarmunora** [comment]
im not sure the hadar really needs the buff, but the code looks good

**Comment by ZeWaka** [comment]
> im not sure the hadar really needs the buff, but the code looks good

you can get bonus points with the players after it becomes too strong and you need to nerf it :)))

**Pull request merged**